src/smalloc.cc

```diff
@@ -111,6 +85,15 @@ CallbackInfo::~CallbackInfo() {
 }
 
 
+void CallbackInfo::DisposeNoAllocation(Isolate* isolate) {
+  HandleScope scope(isolate);
+  Local<Object> object = PersistentToLocal(isolate, persistent_);
+  void* array_data = object->GetIndexedPropertiesExternalArrayData();
+  free(array_data);
```

Comment:
This is called when environment is being destroyed (and thus the isolate), it wouldn't matter.